lib/github_changelog_generator/generator/generator_tags.rb

```diff
@@ -119,41 +119,29 @@ def get_filtered_tags(all_tags)
     # @param [Array] all_tags all tags
     # @return [Array] filtered tags according :since_tag option
     def filter_since_tag(all_tags)
-      filtered_tags = all_tags
-      tag = since_tag
-      if tag
-        if all_tags.map { |t| t["name"] }.include? tag
-          idx = all_tags.index { |t| t["name"] == tag }
-          filtered_tags = if idx
-                            all_tags[0..idx]
-                          else
-                            []
-                          end
-        else
-          raise ChangelogGeneratorError, "Error: can't find tag #{tag}, specified with --since-tag option."
-        end
+      return all_tags unless (tag = since_tag)
+
+      raise ChangelogGeneratorError, "Error: can't find tag #{tag}, specified with --since-tag option." if all_tags.none? { |t| t["name"] == tag }
+
+      if (idx = all_tags.index { |t| t["name"] == tag })
+        all_tags[0..idx]
+      else
+        []
       end
-      filtered_tags
     end
 
     # @param [Array] all_tags all tags
     # @return [Array] filtered tags according :due_tag option
     def filter_due_tag(all_tags)
-      filtered_tags = all_tags
-      tag           = due_tag
-      if tag
-        if all_tags.any? && all_tags.map { |t| t["name"] }.include?(tag)
-          idx = all_tags.index { |t| t["name"] == tag }
-          filtered_tags = if idx > 0
-                            all_tags[(idx + 1)..-1]
-                          else
-                            []
-                          end
-        else
-          raise ChangelogGeneratorError, "Error: can't find tag #{tag}, specified with --due-tag option."
-        end
+      return all_tags unless (tag = due_tag)
```

Comment:
This here is a very compact fix to the snarl of conditionals in the before. Kudos.